ocaml/xapi/repository_helpers.ml

```diff
@@ -1079,3 +1079,24 @@ let with_access_token ~token ~token_id f =
   | _ ->
       let msg = Printf.sprintf "%s: The token or token_id is empty" __LOC__ in
       raise Api_errors.(Server_error (internal_error, [msg]))
+
+let sort_in_assoc x =
+  Yojson.Basic.Util.to_assoc x
```

Comment:
What does the sorting accomplish? It looks like the core of the task is to filter out livepatches. Why does this require sorting?

Reply:
Hm. This is to make it work with the built-in equality. I think it would be better to have an OCaml record type that defines a live patch and use equality over that to make it more explicit. The whole quality of JSON assocs is a bit too indirect for my taste.